### PR TITLE
Check whether a conditional dependency has already been activated before scheduling it for processing

### PR DESCRIPTION
**Pull request by aloubyansky**
@Ladicek this will fix your issue with Vert.X Web Sessions.

### CODE REVIEW

**Comment by Ladicek** [comment]
I think I found a better way that works without this fix, but thanks!

**Comment by aloubyansky** [comment]
No problem, the fix still makes sense though. I added a test too.

**Comment by Ladicek** [comment]
Oh I absolutely agree!

File: independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyTreeResolver.java [hunk 1]
```diff
@@ -332,8 +332,8 @@ private void visitRuntimeDependency(DependencyNode node) {
         final byte prevWalkingFlags = walkingFlags;
         final ExtensionDependency prevLastVisitedRtExtNode = lastVisitedRuntimeExtNode;
 
-        final boolean popExclusions;
-        if (popExclusions = !node.getDependency().getExclusions().isEmpty()) {
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyTreeResolver.java:336]: No real reason

**Review by ia3andy — APPROVED** [review]
Looks good, added a few questions

**Pull request merged**